front/ui/ui-core/src/components/inputs/ComboBox/ComboBox.tsx

```diff
@@ -194,30 +211,39 @@ const ComboBox = <T,>({
         <ul
           className="suggestions-list"
           data-testid={testIdPrefix ? `${testIdPrefix}-list` : undefined}
+          onMouseLeave={() => setActiveSuggestionIndex(-1)}
         >
-          {suggestions.map((suggestion, index) => (
-            <li
-              ref={(el) => {
-                if (suggestionRefs) {
-                  suggestionRefs.current[index] = el;
-                }
-              }}
-              key={`${getSuggestionLabel(suggestion)}-${index}`}
-              data-testid={testIdPrefix ? `${testIdPrefix}-item` : undefined}
-              className={cx('suggestion-item', {
-                active: index === activeSuggestionIndex,
-                selected: value && getSuggestionLabel(value) === getSuggestionLabel(suggestion),
-                small,
-              })}
-              onClick={() => selectSuggestion(index)}
-              onMouseDown={(e) => e.preventDefault()} // Prevents the div parent (.ui-combo-box) from losing focus
-              onMouseEnter={() => {
-                setActiveSuggestionIndex(index);
-              }}
-            >
-              {getSuggestionLabel(suggestion)}
-            </li>
-          ))}
+          {suggestions.map((suggestion, index) => {
+            const isActive = index === activeSuggestionIndex;
+            const isSelected =
+              !!value && getSuggestionLabel(value) === getSuggestionLabel(suggestion);
+            return (
+              <li
+                ref={(el) => {
+                  if (suggestionRefs) {
+                    suggestionRefs.current[index] = el;
+                  }
+                }}
+                key={`${getSuggestionLabel(suggestion)}-${index}`}
+                data-testid={testIdPrefix ? `${testIdPrefix}-item` : undefined}
+                className={cx('suggestion-item', {
+                  active: index === activeSuggestionIndex,
+                  selected: value && getSuggestionLabel(value) === getSuggestionLabel(suggestion),
+                  small,
+                  'suggestion-item--custom': listElementComponent,
```

Comment:
Hum here you are giving a function, it would be better to have a boolean. Maybe this would be clearer 
```suggestion
                  'suggestion-item--custom': !!listElementComponent,
```